### PR TITLE
interface: added graph reducer, store state, and subscription

### PR DESCRIPTION
**Pull request by tacryt-socryp**
Works fine and has no functionality changes. Will append and test scrying functionality to this PR, then we can get it merged with no functionality changes. Will make it easier to review the Links migration as its own thing.

### CODE REVIEW

**Comment by matildepark** [comment]
Looks nice and clean so far, but I'll wait for the remainder too.

**Comment by tacryt-socryp** [comment]
@belisarius222 is there an efficient way to convert between a mop and a map?

**Comment by belisarius222** [comment]
No.  It might be good to add conversion routines from mop to map and back, although arguably converting to list as an intermediate is ok.

**Comment by tacryt-socryp** [comment]
@matildepark All scries have been tested to work. The reducer isn't finished, but far enough that I can move to Links interface integration.

File: pkg/interface/src/logic/reducers/graph-update.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import _ from 'lodash';
+
+
+const OrderedMap = (arr = []) => {
+  let map = new Map(arr);
+//  map[Symbol.iterator] = function* () {
```
Comment (belisarius222) [inline pkg/interface/src/logic/reducers/graph-update.js:6]: Why is this commented out?

**Comment by matildepark** [comment]
> Will append and test scrying functionality to this PR, then we can get it merged with no functionality changes.

Wasn't the scope of this PR intentionally limited?

**Comment by tacryt-socryp** [comment]
Yep! Good point. Removed that commit. This is good to go. I'd like to merge it into release/next-userspace now.

**Comment by matildepark** [comment]
Merge #3352 while you're at it. I'm gonna review the s3 one now ...

**Pull request merged**